src/lib/logical_query_plan/lqp_expression.hpp

```diff
@@ -37,5 +46,6 @@ class LQPExpression : public AbstractExpression<LQPExpression> {
 
  private:
   std::optional<LQPColumnReference> _column_reference;
+  std::optional<std::shared_ptr<AbstractLQPNode>> _subselect_node;
```

Comment:
I don't know if there is some overall consensus on this, but I think wrapping a nullable pointer into an optional might take it too far. If used without care, there are even two representations for an absent value ("absent" and "optional(nullptr)").